### PR TITLE
lesson-3 homework 22.40 01.09.2021

### PR DESCRIPTION
**Pull request by olennikovandrey**

### CODE REVIEW

File: js/main.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+    let result = ''; //declare a variable for string value of result
+    for (let i = 0; i < stringLen; i++) { //set a condition
+        if (string[i] === string[stringLen - 1 - i]) { //Compare all elements from the beginning of the line with all elements from the end of the line
+        result = 'yes'; //if the values ​​are strictly equal, then output the value 'yes'
+        } else {
+        result = 'no'; //else - 'no'
+        return result;
+        }
+    }
+    return result;
```
Comment (Dronom) [inline js/main.js:60]: use true/false

Comment (olennikovandrey) [reply js/main.js:60]: fixed

File: js/main.js [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    for (let i = 0; i < stringLen; i++) { //set a condition
+        if (string[i] === string[stringLen - 1 - i]) { //Compare all elements from the beginning of the line with all elements from the end of the line
+        result = 'true'; //if the values ​​are strictly equal, then output the value 'true'
+        } else {
+        result = 'false'; //else - 'false'
+        return result;
+        }
+    }
+    return result;
+}
```
Comment (Dronom) [inline js/main.js:61]: you can write:
```
if (string[i] === string[stringLen - 1 - i]) {
  continue;
} else {
  return false;
  }
}
return true;
}

```

Comment (olennikovandrey) [reply js/main.js:61]: done, thank you

File: js/lesson-2-3.js [hunk 3]
```diff
@@ -20,53 +20,52 @@ console.log(test); //the result is 20, because the numerical value "1" is subtra
...
     }
 }
 
-console.log('lesson_3_homework')
-//point 5 of homework 'lesson-3'
+console.log('lesson_3_homework');
+console.log('point_5_of_homework_lesson_3');
 function palindrome(string) { //declare a function
     let stringLen = string.length; //declare a variable for test word
     let result = ''; //declare a variable for string value of result
```
Comment (Dronom) [inline js/lesson-2-3.js:51]: you don't need this variable more

File: js/lesson-2-3.js [hunk 4]
```diff
@@ -20,53 +20,52 @@ console.log(test); //the result is 20, because the numerical value "1" is subtra
...
 let second_arr = [2, 5, 8, 15, 0, 6, 20, 3];
 for (let i = 0; i < second_arr.length; i++) {
     if (second_arr[i] > 5 && second_arr[i] < 10) {
         console.log(second_arr[i]);
     }
 }
 
-//point 10 of homework
+console.log('point_10_of_homework_lesson_2');
 let third_arr = [2, 5, 8, 15, 0, 6, 20, 3];
```
Comment (Dronom) [inline js/lesson-2-3.js:40]: is the variable will be reinitialized? I don't think so, use const

Comment (olennikovandrey) [reply js/lesson-2-3.js:40]: fixed)

File: js/lesson-2-3.js (hunk 3 above)
Comment (olennikovandrey) [reply js/lesson-2-3.js:51]: fixed!

File: js/lesson-6.js [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+const result = hotels.filter(item => Object.values(item).some(val => val.toLowerCase().trim().includes(searchValue)));
+
+console.log(result);*/
+
+/*console.log('\tpoint 7');
+const uniqueValues = hotels.filter(unique => unique(`${hotels[i].country} !== ${hotels[i].country} ${hotels[i].city} !== ${hotels[i].city} ${hotels[i].name} !== ${hotels[i].name}`)); {
+    uniqueValues.push(hotels[i].country, hotels[i].city, hotels[i].name);
+}
+
+console.log(uniqueValues);*/
```
Comment (Dronom) [inline js/lesson-6.js:185]: remove unnecessary comments

File: js/lesson-6.js [hunk 6]
```diff
@@ -0,0 +1,185 @@
+console.log('lesson 6 homework');
+console.log('\tpoint 5');
+
+const palindrome = (word) => {
+    debugger
```
Comment (Dronom) [inline js/lesson-6.js:5]: remove debugger from PR and don't forget about ;

File: __tests__/lesson-4.test.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+test("sum function"), () => {
+expect(sum(5, 2)).toBe(7);
+}
+
+
+/*
+
+test('sum 5 + 2 equals 7', () => {
+    expect(sum(5)(2)).toBe(7);
+});*/
```
Comment (Dronom) [inline __tests__/lesson-4.test.js:12]: remove

[the remark this answers is not in the document]
Comment (olennikovandrey) [reply __tests__/lesson-4.test.js]: done

**Pull request merged**